scripts/linters/app_dev_linter.py

```diff
@@ -0,0 +1,79 @@
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Lint checks for skip_files in app.yaml file."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import glob
+import os
+import sys
+
+import python_utils
+from . import linter_utils
+
+APP_YAML_FILEPATH = os.path.join(os.getcwd(), 'app_dev.yaml')
+
+
+def check_skip_files_in_app_dev_yaml(file_cache, verbose_mode_enabled):
+    """Check to ensure that all lines in skip_files in app.yaml reference valid
+    files in the repository.
+    """
+    if verbose_mode_enabled:
+        python_utils.PRINT('Starting app_dev file check')
+        python_utils.PRINT('----------------------------------------')
+
+    with linter_utils.redirect_stdout(sys.stdout):
+        failed = False
+        summary_messages = []
+        skip_files_section_found = False
+        for line_num, line in enumerate(file_cache.readlines(
+                APP_YAML_FILEPATH)):
+            stripped_line = line.strip()
+            if '# Third party files:' in stripped_line:
+                skip_files_section_found = True
+            if not skip_files_section_found:
+                continue
+            if stripped_line and stripped_line[0] != '#':
+                # Extract the file pattern from the line.
+                line_in_concern = stripped_line[2:]
+                # Adjustments to the dir paths in app_dev.yaml file
+                # for glob-style patterns to match correctly.
+                if line_in_concern.endswith('/'):
+                    line_in_concern = line_in_concern[:-1]
+                if not glob.glob(line_in_concern):
+                    summary_message = (
+                        '%s --> Pattern on line %s doesn\'t match '
+                        'any file or directory' % (
+                            APP_YAML_FILEPATH, line_num + 1))
+                    summary_messages.append(summary_message)
+                    python_utils.PRINT(summary_message)
+                    failed = True
+
+        if failed:
+            summary_message = (
+                '%s   app_dev file coverage check failed, see messages above '
+                'for invalid file names in app_dev.yaml file' %
+                linter_utils.MESSAGE_TYPE_FAILED)
+        else:
+            summary_message = '%s   app_dev file check passed' % (
+                linter_utils.MESSAGE_TYPE_SUCCESS)
+        summary_messages.append(summary_message)
+        python_utils.PRINT(summary_message)
```

Comment:
is this a somewhat duplicate of line 64, if yes then print it once!

Reply:
No, we need both lines

Reply:
Only have that problem with pylint.

Reply:
I think there could be problem that Sandeep mentioned that is messages could come up between these two print statements.